src/components/Reports/GoalCalculator/GoalCard/GoalCard.tsx

```diff
@@ -65,113 +73,168 @@ const StyledActionBox = styled(Box)({
-        <StyledContentInnerBox>
-          <StyledInfoRow pb={theme.spacing(1)}>
-            <Typography variant="body1" fontWeight="bold" pl={2}>
-              {t('Goal Amount')}
-            </Typography>
-            <Typography data-testid="goal-amount-value" variant="body1">
-              {currencyFormat(goalAmount, 'USD', locale)}
+    <>
+      <Confirmation
+        title={t('Delete Goal')}
+        isOpen={deleting}
+        mutation={handleConfirmDelete}
+        handleClose={handleCancelDelete}
+        confirmLabel={t('Delete Goal')}
+        cancelLabel={t('Cancel')}
+        message={
+          <Trans t={t}>
+            Are you sure you want to delete <strong>{goal.createdAt}</strong>?
+            Deleting this goal will remove it permanently.
+          </Trans>
+        }
+        confirmButtonProps={{
+          variant: 'contained',
+          color: 'error',
+          children: t('Delete Goal'),
+        }}
+      />
+
+      <StyledCard>
+        <StyledHeaderBox>
+          <StyledTitleBox>
+            <Typography data-testid="goal-title" variant="h6">
+              {goal.createdAt}
             </Typography>
-          </StyledInfoRow>
-
-          <Divider />
-
-          <StyledInfoRow pt={theme.spacing(1)}>
-            <Typography variant="body1" fontWeight="bold" pl={2}>
-              {t('Last Updated')}
-            </Typography>
-            <Typography data-testid="date-value" variant="body1">
-              {goalDate.toLocaleString({
-                month: 'long',
-                day: 'numeric',
-                year: 'numeric',
-              })}
+          </StyledTitleBox>
+          {renderStar && (
+            <StyledStarButton
+              aria-label="star-button"
+              onClick={handleStarClick}
+            >
+              {goal.primary ? (
+                <Star color="primary" sx={{ verticalAlign: 'middle' }} />
+              ) : (
+                <StarBorderOutlined
+                  color="primary"
+                  sx={{ verticalAlign: 'middle' }}
+                />
+              )}
+            </StyledStarButton>
+          )}
+        </StyledHeaderBox>
+
+        <Divider />
+
+        <StyledContentBox>
+          <StyledContentInnerBox>
+            <StyledInfoRow pb={theme.spacing(1)}>
+              <Typography variant="body1" fontWeight="bold" pl={2}>
+                {t('Goal Amount')}
+              </Typography>
+              <Typography data-testid="goal-amount-value" variant="body1">
+                {currencyFormat(0, 'USD', locale)}
+              </Typography>
+            </StyledInfoRow>
+
+            <Divider />
+
+            <StyledInfoRow pt={theme.spacing(1)}>
+              <Typography variant="body1" fontWeight="bold" pl={2}>
+                {t('Last Updated')}
+              </Typography>
+              <Typography data-testid="date-value" variant="body1">
+                {dateFormat(DateTime.fromISO(goal.createdAt), locale, {
```

Comment:
[Nice-to-have] Excellent use of shared dateFormat utility with fullMonth option. Evidence: `dateFormat(DateTime.fromISO(goal.createdAt), locale, { fullMonth: true })`. Impact: Follows repo conventions perfectly and improves consistency. [Category: File]